main.go

```diff
@@ -448,7 +448,17 @@ func newWatcher(kapi client.KeysAPI, path string, socksProxy string, etcdPeers [
 		for {
 			_, err := watcher.Next(context.Background())
 			if err != nil {
-				log.Printf("watch failed %v, sleeping for 1s\n", err.Error())
+				if err == context.Canceled {
+					log.Println("context cancelled error")
+				} else if err == context.DeadlineExceeded {
+					log.Println("deadline exceeded error")
```

Comment:
These two are admittedly not great, but those are pretty unexpected and uninteresting cases, it's the ones below (especially ClusterError) that I'm really more interested in.